Wrappers/Python/ccpi/viewer/CILViewer.py

```diff
@@ -835,7 +853,16 @@ def updateVolumePipeline(self):
             opacity = colormaps.CILColorMaps.get_opacity_transfer_function(x, 
             colormaps.relu, cmin, cmax, scaling)
             self.volume_property.SetColor(colors)
-            self.volume_property.SetScalarOpacity(opacity)
+
+            # Update whether we use our calculated opacity as the scalar or gradient opacity
+            # Also return the other opacity type to its default value:
+            if self.getVolumeRenderOpacityMethod() == 'gradient':
+                self.volume_property.SetScalarOpacity(self.getDefaultScalarOpacity()) 
+                self.volume_property.SetGradientOpacity(opacity)
+                 
+            elif self.getVolumeRenderOpacityMethod() == 'scalar': 
+                self.volume_property.SetGradientOpacity(self.getDefaultGradientOpacity())
+                self.volume_property.SetScalarOpacity(opacity)
         
 
     def setVolumeColorLevelWindow(self, cmin, cmax):
```

Comment:
Is the intention that this could be set by the user?
would it make more sense for the user to set the percentiles? I am not sure